src/molecules/ChipLarge.tsx

```diff
@@ -12,39 +22,68 @@ const styles = StyleSheet.create({
     borderRadius: 11,
     height: 22,
   } as ViewStyle,
-  IconAndText: {
-    flexDirection: 'row',
-    alignItems: 'center',
-  },
   Icon: {
     marginRight: 4,
   },
   Text: registeredTextStyle('titleBold', colors.ulisse, 'chip-text'),
   TextDark: registeredTextStyle('titleBold', colors.uma, 'chip-text-dark'),
 });
 
+type containerStyle = {
+  paddingLeft?: FlexStyle['paddingLeft'];
+  paddingRight?: FlexStyle['paddingRight'];
+  paddingTop?: FlexStyle['paddingTop'];
+  paddingBottom?: FlexStyle['paddingBottom'];
+  marginLeft?: FlexStyle['marginLeft'];
+  marginRight?: FlexStyle['marginRight'];
+  marginTop?: FlexStyle['marginTop'];
+  marginBottom?: FlexStyle['marginBottom'];
+};
+
 export type ChipLargeProps = {
   label: string;
-  icon?: string;
+  icon?: ImageRequireSource | string;
   color: string;
   /**
    * If specified, forces text to be dark (can be used to skip color lightness computation
    * if there are many chips on screen)
    */
   colorIsLight?: boolean;
   onPress?: () => any;
+  containerStyle?: containerStyle;
+};
+
+const renderImageOrIcon = (
+  image: ImageRequireSource | string,
+  darkText: boolean,
+  style: StyleProp<ViewStyle>
+) => {
+  switch (image) {
+    case 'walk':
+      return <Icon name={image} size={20} color={colors.uto} />;
+  }
```

Comment:
an `if` would be more readable, with just one case

Reply:
Thanks. Commit https://github.com/urbi-mobility/react-native-urbi-ui/commit/22da099533b87809da2e5e2ede52d817f6221fcd